rfc/patchhost.go

```diff
@@ -0,0 +1,10 @@
+package rfc
+
+import "strings"
+
+// PatchHost returns a host string without trailing dot. For details
+// see also the discussion in
+// https://lists.w3.org/Archives/Public/ietf-http-wg/2016JanMar/0430.html.
+func PatchHost(host string) string {
+	return strings.TrimSuffix(host, ".")
```

Comment:
https://pkg.go.dev/net/http#Request
> It may be of the form "host:port"

Reply:
Maybe?
```go
	host = strings.Replace(host, ".:", ":", -1)
	host = strings.TrimSuffix(host, ".")
```

Reply:
yes thanks there was also more wrong because of outgoinghost..., now it's fixed in all cases (tested with curl and nc)